extensions/omni_python/docs/intro.md

```diff
@@ -0,0 +1,272 @@
+
+For example, let's make two Python files.
+
+`hello.py`:
+```python
+from omni_python import pg
+
+@pg
+def hello() -> str:
+    return "Hey there!"
+```
+
+`maths.py`:
+```python
+from omni_python import pg
+
+@pg
+def add(a: int, b: int) -> int:
+    return a + b
+
+@pg
+def subtract(a: int, b: int) -> int:
+    return a - b
+```
+Create a function to define an identifier for the local directory you want to load into the database. For the example, the local directory is `python-files`.
+
+```postgresql
+create or replace function demo_function() returns omni_vfs.local_fs language sql
+as $$
+select omni_vfs.local_fs('/python-files')
+$$;
+```
+Configure `omni_python`
+```postgresql
+insert into omni_python.config (name, value) values ('pip_find_links', '/python-wheels');
+```
+!!! tip "Tip"
+
+    We are working on a CLI tooling that will take care of directory mapping.
+
+Load the filesystem files.
+```postgresql
+select omni_schema.load_from_fs(demo_function());
+```
+
+!!! tip "Optional Tip"
+
+    You can set a reload command for reloading the filesystem changes.
+    ```postgresql
+    \set reload 'select omni_schema.load_from_fs(demo_function());'
+    ```
+
+
+Run Omnigres in a Docker container. Make sure to mount the local directory as a volume on the correct path.
+```shell
+docker run --name omnigres \
+           -e POSTGRES_PASSWORD=omnigres \
+           -e POSTGRES_USER=omnigres \
+           -e POSTGRES_DB=omnigres \
+           --mount source=omnigres,target=/var/lib/postgresql/data -v $(pwd)/python-files:/python-files \
+           -p 127.0.0.1:5433:5432 --rm ghcr.io/omnigres/omnigres-slim:latest
+```
+
+Let's try it out!
+```postgresql
+omnigres=# select hello();
+   hello
+------------
+ Hey there!
+```
+```postgresql
+omnigres=# select add(5, 10);
+ add
+-----
+  15
+```
+```postgresql
+omnigres=# select subtract(5, 10);
+ subtract
+----------
+       -5
+```
+
+## Flask
+
+For Flask framework integration, we have a few more steps.
+
+Create `omni_httpd` extension to be able to handle HTTP requests.
+```postgresql
+create extension if not exists omni_httpd cascade;
+```
+
+Add the following in `requirements.txt` file.
+```
+omni_http[Flask]
+```
+
+Create HTTP listeners for our Flask application. For example, let's add a HTTP listener for port 5000.
+```postgresql
+with
+    listener as (insert into omni_httpd.listeners (address, port) values ('0.0.0.0', 5000) returning id),
+    handler as (insert into omni_httpd.handlers (query) values
+                                                 (
+                                                 $$ select handle(request.*) from request$$
+       ) returning id)
+insert
+into
+    omni_httpd.listeners_handlers (listener_id, handler_id)
+select
+    listener.id,
+    handler.id
+from
+    listener,
+    handler;
+```
+
+Let's say you have a table called `employees`.
+```postgresql
+create table employees (
+   id uuid,
```

Comment:
Minor nitpicking:

I/we generally don't recommend `uuid` for the `id` columns.

Throught out the project we use

```sql
 id                       integer primary key generated always as identity,
```

For user applications, it's often recommended to bump `int` to `int8` for practical purposes:

```sql
 id8                       integer primary key generated always as identity,
```

The idea here is to teach good practices consistently.

Reply:
> I/we generally don't recommend uuid for the id columns.

Just curious about this: Why?

Reply:
Along these lines:

* https://medium.com/@sjksingh/postgresql-primary-key-dilemma-uuid-vs-bigint-52008685b744
* https://pganalyze.com/blog/5mins-postgres-uuid-vs-serial-primary-keys

Reply:
Thanks, will check!